### PR TITLE
Added various command-line options

### PR DESCRIPTION
**Pull request by JeromSar**
Created a foundation for adding more options.
Currently implemented: --port, --host and --onlinemode

Some server hosters, especially those who have payment options on a per-slot basis, customize runtime parameters to restrict server owners from exceeding their limits. Glowstone currently has limited support for command line options.

This commit adds a foundation for command-line options and implements --host, --port and --onlinemode

This commit also formats the help message that it shown when running GlowStone with the --help switch.

Lastly, some code cleanup and refractoring is also included in this commit.

### CODE REVIEW

**Comment by SpaceManiac** [comment]
I'll accept the `{}` and `final` formatting changes, but you remove and add blank lines in several weird places. Ideally those would be left unchanged.

**Comment by JeromSar** [comment]
@SpaceManiac Those were automatic formats. I'll change them back for you. :)

File: src/main/java/net/glowstone/GlowServer.java [hunk 1]
```diff
@@ -76,61 +77,102 @@ public static void main(String[] args) {
...
+            // Version
+            if ("--version".equals(opt) || "-v".equals(opt)) {
+                System.out.println("Glowstone version: " + GlowServer.class.getPackage().getImplementationVersion());
+                System.out.println("Bukkit version:    " + GlowServer.class.getPackage().getSpecificationVersion());
+                System.out.println("Minecraft version: " + GAME_VERSION + " protocol " + PROTOCOL_VERSION);
+                return null;
+            }
+
+            // Below this point, options require parameters
+            if (args.length == i) {
```
Comment (SpaceManiac) [inline src/main/java/net/glowstone/GlowServer.java]: Want `i == args.length - 1` or similar.

Comment (JeromSar) [inline src/main/java/net/glowstone/GlowServer.java]: Woops didn't see that.

File: src/main/java/net/glowstone/GlowServer.java [hunk 2]
```diff
@@ -706,7 +752,6 @@ public boolean dispatchCommand(CommandSender sender, String commandLine) throws
 
     ////////////////////////////////////////////////////////////////////////////
     // Player management
-
```
Comment (SpaceManiac) [inline src/main/java/net/glowstone/GlowServer.java:709]: Wonky whitespace change still.

File: src/main/java/net/glowstone/util/ServerConfig.java [hunk 3]
```diff
@@ -40,10 +42,27 @@
...
+        Validate.notNull(parameters);
+
+        this.configDir = configDir;
+        this.configFile = configFile;
+        this.parameters = parameters;
+    }
+
+    /**
      * Initialize a new ServerConfig and associated settings.
+     *
```
Comment (SpaceManiac) [inline src/main/java/net/glowstone/util/ServerConfig.java]: Whitespace change.

File: src/main/java/net/glowstone/util/ServerConfig.java [hunk 4]
```diff
@@ -162,15 +194,15 @@ public void load() {
...
         if (resource == null) {
             GlowServer.logger.warning("Could not find default " + source + " on classpath");
             return;
         }
 
-        try (InputStream in = resource.openStream();
-             OutputStream out = new FileOutputStream(dest)) {
-            byte[] buf = new byte[2048];
+        try (final InputStream in = resource.openStream();
+                final OutputStream out = new FileOutputStream(dest)) {
```
Comment (SpaceManiac) [inline src/main/java/net/glowstone/util/ServerConfig.java:203]: ???

Comment (JeromSar) [inline src/main/java/net/glowstone/util/ServerConfig.java:203]: @SpaceManiac final, final

Comment (SpaceManiac) [inline src/main/java/net/glowstone/util/ServerConfig.java:203]: oh I see, I was confused by your spacing because of the try. actually fine I think.

**Comment by JeromSar** [comment]
All tested, of course. :)

**Comment by SpaceManiac** [comment]
Merged in b66684a054e47c5019c8d6c12bc7430e8cf973b3.

**Pull request closed without merging**